src/services/TechRecordsService.ts

```diff
@@ -185,6 +178,48 @@ class TechRecordsService {
         throw new HTTPError(500, HTTPRESPONSE.INTERNAL_SERVER_ERROR);
       });
   }
+
+  public updateVin(vehicle: Vehicle, newVin: string) {
+    const vehicleClone = cloneDeep(vehicle);
+
+    const oldVehicle: Vehicle = { ...vehicleClone, techRecord: [] };
+    const newVehicle: Vehicle = {
+      ...vehicleClone,
+      vin: newVin,
+      techRecord: [],
+    };
+    let current: TechRecord | undefined;
+    let provisional: TechRecord | undefined;
+
+    vehicleClone.techRecord.forEach((record) => {
+      switch (record.statusCode) {
+        case STATUS.PROVISIONAL:
+          provisional = { ...record };
+          break;
+        case STATUS.CURRENT:
+          current = { ...record };
+          break;
+        case STATUS.ARCHIVED:
+          oldVehicle.techRecord.push({ ...record });
+          break;
+        default:
+          break;
+      }
+    });
+
+    if (current && provisional) {
+      newVehicle.techRecord.push({ ...current });
+      oldVehicle.techRecord.push({ ...current, statusCode: "archived" });
+    } else if (provisional) {
+      newVehicle.techRecord.push({ ...provisional });
+      oldVehicle.techRecord.push({ ...provisional, statusCode: "archived" });
+    } else if (current) {
+      newVehicle.techRecord.push({ ...current });
+      oldVehicle.techRecord.push({ ...current, statusCode: "archived" });
+    }
+
```

Comment:
Could we not make this whole section a bit condensed? Do the processing in the loop? Not sure it matters tho, this is still good and readable.

Reply:
^ shut up nathan